src/application/environment.rs

```diff
@@ -0,0 +1,71 @@
+
+#[derive(Debug, Clone)]
+pub struct EffiEnvironment {
+    pub todo_directory_path: String,
+    pub todo_file_path: String,
+    pub done_file_path: String
+}
+
+fn initialize_effi_file(file_path: &str){
+    // If a todo.txt or done.txt file does not exist, create an empty one
+    if !std::fs::metadata(file_path).is_ok(){
+        log::info!("todo.txt or done.txt does not exist, creating an empty file at {}", file_path);
+        match std::fs::File::create(file_path){
+            Ok(_) => {},
+            Err(err) => {
+                log::error!("Unable to create {} because {}",file_path, err);
```

Comment:
Errors shouldn’t be treated here but reported to caller.

Reply:
I am reporting these errors back up with a ```Result<(), std::io::Error>``` to the ```Environment``` constructor now, and the constructor logs an error if there are any initialization problems. I'll hold off on resolving this in case we need to bubble this up any further

Reply:
The constructor can return a `Result` and let the main function deals with error.

Reply:
Thanks for the feedback, will work on this when I get a chance...if it were to return an Error to the caller in ```main()``` should we panic at that point? Or is there someway to recover from this (fallback todo.txt and done.txt for example)?

Reply:
Unwrapping this in main, let me know your thoughts though